Lib9c/BlockPolicySource.cs

```diff
@@ -87,21 +87,31 @@ bool ignoreHardcodedIndicesForBackwardCompatibility
         public IEnumerable<IRenderer<NCAction>> GetRenderers() =>
             new IRenderer<NCAction>[] { BlockRenderer, LoggedActionRenderer };
 
+        public static bool IsObsolete(Transaction<NCAction> transaction)
+        {
+            return transaction.Actions
+                .Select(action => action.InnerAction.GetType())
+                .Any(at => at.IsDefined(typeof(ObsoleteAttribute), false));
```

Comment:
It can be a matter because checking obsoleted depends on block height(index). so we need to specify its validity period on an attribute to check here.